internal/credentials/spiffe/spiffe.go

```diff
@@ -81,3 +83,29 @@ func BundleMapFromBytes(bundleMapBytes []byte) (BundleMap, error) {
 	}
 	return bundleMap, nil
 }
+
+// GetRootsFromSPIFFEBundleMap returns the root trust certificates from the
+// SPIFFE bundle map for the given trust domain from the leaf certificate.
+func GetRootsFromSPIFFEBundleMap(bundleMap BundleMap, leafCert *x509.Certificate) (*x509.CertPool, error) {
+	// 1. Upon receiving a peer certificate, verify that it is a well-formed SPIFFE
+	//    leaf certificate.  In particular, it must have a single URI SAN containing
+	//    a well-formed SPIFFE ID ([SPIFFE ID format]).
+	spiffeId := credinternal.SPIFFEIDFromCert(leafCert)
```

Comment:
Concerned about using this method here, discussing offline.

Reply:
Fixed